chain/client/src/chunk_validation.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::Arc;
+
+use near_async::messaging::{CanSend, Sender};
+use near_chain::types::RuntimeAdapter;
+use near_chain_primitives::Error;
+use near_epoch_manager::EpochManagerAdapter;
+use near_network::types::{NetworkRequests, PeerManagerMessageRequest};
+use near_primitives::checked_feature;
+use near_primitives::chunk_validation::{
+    ChunkEndorsement, ChunkEndorsementInner, ChunkEndorsementMessage, ChunkStateWitness,
+};
+use near_primitives::sharding::ShardChunkHeader;
+use near_primitives::types::EpochId;
+use near_primitives::validator_signer::ValidatorSigner;
+
+use crate::Client;
+
+/// A module that handles chunk validation logic. Chunk validation refers to a
+/// critical process of stateless validation, where chunk validators (certain
+/// validators selected to validate the chunk) verify that the chunk's state
+/// witness is correct, and then sends chunk endorsements to the block producer
+/// so that the chunk can be included in the block.
+pub struct ChunkValidator {
+    my_signer: Option<Arc<dyn ValidatorSigner>>,
```

Comment:
what does 'my_signer' mean? :o

Reply:
If I'm a validator, this is my signer. It's the signer of the node itself. I'll make a comment.